flink-connectors/flink-connector-kafka/src/test/java/org/apache/flink/streaming/connectors/kafka/table/ReducingUpsertWriterTest.java

```diff
@@ -297,6 +432,26 @@ public Long timestamp() {
         }
     }
 
+    private void writeDataWithNullTimestamp(
+            ReducingUpsertWriter<?> writer, Iterator<RowData> iterator) throws Exception {
+        while (iterator.hasNext()) {
+            RowData next = iterator.next();
+            writer.write(
```

Comment:
You can just call this once and invoke flush() afterwards. There's even no need to test the buffering logic as that is captured by other tests.